pkg/chunk/cassandra/storage_client.go

```diff
@@ -44,6 +47,7 @@ type Config struct {
 	MinBackoff               time.Duration       `yaml:"retry_min_backoff"`
 	QueryConcurrency         int                 `yaml:"query_concurrency"`
 	NumConnections           int                 `yaml:"num_connections"`
+	ConvictHosts             bool                `yaml:"convict_hosts"`
```

Comment:
For clarity, I would call the YAML option/CLI flag `convict_hosts_on_failure`/`convict-hosts-on-failure`